app/views/handlers/question.py

```diff
@@ -254,3 +262,44 @@ def clear_radio_answers(self):
                 answer_ids_to_remove, self.current_location.list_item_id
             )
             self.questionnaire_store_updater.save()
+
+    def get_first_incomplete_repeating_block_location(
+        self, repeating_block_ids: Sequence[str], section_id: str, list_name: str
+    ) -> Location | None:
+        if not repeating_block_ids:
+            return None
+
+        list_model = self._questionnaire_store.list_store.get(list_name)
+        for list_item_id in list_model.items:
+            if incomplete_location := self.get_first_incomplete_repeating_block_location_for_list_item(
+                repeating_block_ids=repeating_block_ids,
+                section_id=section_id,
+                list_item_id=list_item_id,
+                list_name=list_name,
+            ):
+                return incomplete_location
+
+    def get_first_incomplete_repeating_block_location_for_list_item(
+        self,
+        repeating_block_ids: Sequence[str],
+        section_id: str,
+        list_item_id: str,
+        list_name: str,
+    ) -> Location | None:
+        if self.questionnaire_store_updater.is_section_complete(
```

Comment:
I know we spoke about this briefly in the session yesterday, but any reason why we can't do this here: `self._questionnaire_store.progress_store.is_section_complete(section_id=section_id, list_item_id=list_item_id)`? Then we wouldn't need the added method in updater

Reply:
Progress store is a private class attribute so could either make it public or keep the forwarding method

Reply:
Exposing as public implies it should be accessed by those who have access to the QuestionnaireStoreUpdater, whereas having the forwarding method limits accessing it to only the correct circumstances

Reply:
Not quite sure what you mean by it being private? We already access progress store via the questionnaire store in this class (for example in the form method), the same way we do with the answer_store and list_store etc. I do mean `questionnaire_store` rather than `questionnaire_store_updater` if that's causing any confusion

Reply:
Ahhh I see! I didn't realise that QuestionnaireStore was available via parent BlockHandler and that ProgressStore was public via that, thanks for spotting will change that

Reply:
Now making use of `ProgressStore` directly via `QuestionnaireStore` 👍